src/basic_bot/commons/recognition_provider.py

```diff
@@ -94,53 +104,59 @@ def stats(cls) -> Dict[str, Any]:
             "last_frame_duration": cls.last_frame_duration,
         }
 
+    @classmethod
```

Comment:
also did a little reorg'ing.  The while loop in `provide_state` was getting a bit large to grok in a glance